lib/compress/huf_compress.c

```diff
@@ -367,67 +367,161 @@ static U32 HUF_setMaxHeight(nodeElt* huffNode, U32 lastNonNull, U32 maxNbBits)
 }
 
 typedef struct {
-    U32 base;
-    U32 curr;
+    U16 base;
+    U16 curr;
```

Comment:
note : accessing a 32-bit values _tends_ to be faster than accessing 16-bit values on modern cpu architectures.

Not sure if this impact is large enough to be measurable though.

Reply:
Interesting, I'm actually seeing better performance when using 16-bit values.

Reply:
Well, there is also a benefit, which is a smaller memory footprint of the associated array, resulting in reduced L1 cache occupation.
So the benefit might outpace the access deficit.

In any case, it's the measurement which matters, and ensuring that the 16-bit has better (or equivalent) speed is the right move.